applications/mne_analyze/libs/anShared/Model/forwardsolutionmodel.h

```diff
@@ -0,0 +1,174 @@
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ * @brief    Declaration of the ForwardSolutionModel Class.
+ *
+ */
+
+#ifndef FORWARDSOLUTIONMODEL_H
+#define FORWARDSOLUTIONMODEL_H
+
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include "../anshared_global.h"
+#include "../Utils/types.h"
+#include "abstractmodel.h"
+
+//=============================================================================================================
+// FORWARD DECLARATIONS
+//=============================================================================================================
+
+namespace MNELIB {
+    class MNEForwardSolution;
+}
+
+//=============================================================================================================
+// DEFINE NAMESPACE ANSHAREDLIB
+//=============================================================================================================
+
+namespace ANSHAREDLIB {
+
+//=============================================================================================================
+class ANSHAREDSHARED_EXPORT ForwardSolutionModel : public AbstractModel
+{
+    Q_OBJECT
+public:
+    typedef QSharedPointer<ForwardSolutionModel> SPtr;              /**< Shared pointer type for ForwardSolutionModel. */
+    typedef QSharedPointer<const ForwardSolutionModel> ConstSPtr;   /**< Const shared pointer type for ForwardSolutionModel. */
+
+public:
+    //=========================================================================================================
+    ForwardSolutionModel(QObject* parent = Q_NULLPTR);
+
+    //=========================================================================================================
+    ForwardSolutionModel(QSharedPointer<MNELIB::MNEForwardSolution> pFwdSolution,
+                         QObject* parent = Q_NULLPTR);
+
+    //=========================================================================================================
+    /**
+     * Returns the number of rows in the model
+     *
+     * @param[in] parent     The parent index.
+     */
+    virtual int rowCount(const QModelIndex &parent = QModelIndex()) const override;
```

Comment:
This is a general comment: In cases of classes which are final in their inheritance scheme, I would only keep the override keyword and remove the virtual keyword.

I do not think that ForwardSolutionModel will be used as a base class. Maybe consider marking the whole class as final.